kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Guilds.kt

```diff
@@ -0,0 +1,43 @@
+package com.kotlindiscord.kord.extensions.utils
+
+import dev.kord.common.entity.Permission
+import dev.kord.common.entity.Permissions
+import dev.kord.core.Kord
+import dev.kord.core.behavior.GuildBehavior
+import dev.kord.core.entity.Member
+import dev.kord.core.entity.channel.GuildChannel
+import dev.kord.core.supplier.EntitySupplyStrategy
+
+/**
+ * Retrieves the member of the bot itself on this [GuildBehavior].
+ *
+ * @see Kord.selfId
+ */
+public suspend fun GuildBehavior.selfMember(): Member = getMember(kord.selfId)
+
+/**
+ * Checks whether the bot has at least [requiredPermissions] in this [GuildChannel].
+ *
+ * @see GuildBehavior.botHasPermissions
+ */
+public suspend fun GuildChannel.botHasPermissions(vararg requiredPermissions: Permission): Boolean =
+    guild.botHasPermissions(this, Permissions(requiredPermissions.asIterable()))
+
+/**
+ * Checks whether the bot globally has at least [requiredPermissions] on this guild.
+ *
+ * @see GuildChannel.botHasPermissions
+ */
+public suspend fun GuildBehavior.botHasPermissions(vararg requiredPermissions: Permission): Boolean =
+    botHasPermissions(null, Permissions(requiredPermissions.asIterable()))
+
+private suspend fun GuildBehavior.botHasPermissions(channel: GuildChannel?, requiredPermissions: Permissions): Boolean {
+    val effectivePermissions =
+        channel?.run { permissionsForMember(selfMember()) }
+        // fast lane: this guild was fetched through rest
+            ?: asGuild().permissions
+            // slow lane: if guild was cached we fetch it to retrieve permissions field
+            ?: withStrategy(EntitySupplyStrategy.rest).asGuild().permissions!!
```

Comment:
We prefer that API method wrappers provided by KordEx adhere to the caching settings provided by the `ExtensibleBotBuilder`, which defaults to the caching REST supplier - wouldn't `asGuild()` already hit cache?

Reply:
> wouldn't asGuild() already hit cache?
Yes that's the issue because if we take a look at Guild.permissions
https://github.com/kordlib/kord/blob/28fab1af5f768d666e763aebe5a6910629602c29/core/src/main/kotlin/entity/Guild.kt#L96-L102

guilds are normally cache upon gateway connect, so the cached version won't have the permissions field, which is why we fetch permissions through rest if it doesn't

Reply:
Wouldn't checking permissions this way be inherently faulty anyway, though? I mean, this isn't going to check what perms the bot's roles have, right?

Reply:
yep you're right